Extension/src/LanguageServer/client.ts

```diff
@@ -807,6 +815,83 @@ export class DefaultClient implements Client {
+                    var paths = compilerDefaults.knownCompilers.map(function (a) { return a.path; });
+                    const index: number = await ui.showSelectCompiler(paths);
+                    if (index < 0) {
+                    }
+                    util.updateTrustedCompilersList(paths[index])
+                    const selectedCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);;
+                    compilerDefaults = selectedCompilerDefaults;
+                    clients.forEach(client => {
+                        if (client instanceof DefaultClient) {
+                            const defaultClient: DefaultClient = <DefaultClient>client;
+                            defaultClient.configuration.CompilerDefaults = compilerDefaults;
+                            defaultClient.configuration.handleConfigurationChange();
+                        }
+                    });
+                } else {
+                    const selectCompiler: string = localize("selectCompiler.string", "Set Compiler");
+                    const value: string | undefined = await vscode.window.showInformationMessage(localize("selectCompiler.message", `You do not have a compiler configured. Unless you set your own configurations, Intellisense may not be functional`), selectCompiler);
+                    if (value === selectCompiler) {
+                        var paths = compilerDefaults.knownCompilers.map(function (a) { return a.path; });
+                        const index: number = await ui.showSelectCompiler(paths);
+                        if (index < 0) {
+                        }
+                        util.updateTrustedCompilersList(paths[index])
+                        const selectedCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);;
+                        compilerDefaults = selectedCompilerDefaults;
+                        clients.forEach(client => {
+                            if (client instanceof DefaultClient) {
+                                const defaultClient: DefaultClient = <DefaultClient>client;
+                                defaultClient.configuration.CompilerDefaults = compilerDefaults;
+                                defaultClient.configuration.handleConfigurationChange();
+                            }
+                        });
+                    }
+                }
+            } else {
+                var paths = compilerDefaults.knownCompilers.map(function (a) { return a.path; });
+                const index: number = await ui.showSelectCompiler(paths);
+                if (index < 0) {
+                }
+                util.updateTrustedCompilersList(paths[index])
+                const selectedCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);;
+                compilerDefaults = selectedCompilerDefaults;
+                clients.forEach(client => {
```

Comment:
```
                clients.forEach(client => {
                    if (client instanceof DefaultClient) {
                        const defaultClient: DefaultClient = <DefaultClient>client;
                        defaultClient.configuration.CompilerDefaults = compilerDefaults;
                        defaultClient.configuration.handleConfigurationChange();
                    }
                });
```

This repeating process could be moved out of each command check statement and placed at the end of the scope `if (!command) {`